pandas/core/nanops.py

```diff
@@ -474,8 +474,7 @@ def nanargmax(values, axis=None, skipna=True):
     """
     Returns -1 in the NA case
     """
-    values, mask, dtype, _ = _get_values(values, skipna, fill_value_typ='-inf',
-                                         isfinite=True)
```

Comment:
I would be ok with allowing this, but it HAS to work with missing values. Everything else does and no reason for this not too. you simply mask, apply function (e.g. argmax), then replace the missing by -1